packages/react/src/SubdomainNavBar/SubdomainNavBar.tsx

```diff
@@ -299,12 +268,61 @@ function Root({
                       .filter(Boolean)}
                   </div>
                 </div>
-              </div>
-            )}
+              )}
+
+              {!isMedium && (
+                <div
+                  className={clsx(
+                    styles['SubdomainNavBar-menu-wrapper'],
+                    menuHidden && styles['SubdomainNavBar-menu-wrapper--close'],
+                  )}
+                >
+                  {hasLinks && !menuHidden && (
+                    <NavigationVisbilityObserver
+                      showOnlyOnNarrow
+                      className={clsx(styles['SubdomainNavBar-primary-nav-list--visible'])}
+                    >
+                      {menuItems}
+                    </NavigationVisbilityObserver>
+                  )}
+                  <div
+                    className={clsx(
+                      styles['SubdomainNavBar-button-area'],
+                      styles['SubdomainNavBar-button-area--visible'],
+                    )}
+                  >
+                    <div className={styles['SubdomainNavBar-button-area-inner']}>
+                      {React.Children.toArray(children)
+                        .map(child => {
+                          if (React.isValidElement(child) && typeof child.type !== 'string') {
+                            if (child.type === PrimaryAction) {
+                              return child
+                            }
+                            return null
+                          }
+                        })
+                        .filter(Boolean)}
+
+                      {React.Children.toArray(children)
+                        .map(child => {
+                          if (React.isValidElement(child) && typeof child.type !== 'string') {
+                            if (child.type === SecondaryAction) {
+                              return child
+                            }
+                            return null
+                          }
+                        })
+                        .filter(Boolean)}
+                    </div>
+                  </div>
+                </div>
+              )}
+            </div>
           </div>
-        </div>
-      </header>
-    </div>
+        </header>
+      </div>
+      <div id="start-of-content" />
```

Comment:
Do avoid any id clashes on end-user side, could we make this id unique please? Some precedents in other components via `useId` hooks.